### PR TITLE
Upgrade from core18 to core20

### PR DESCRIPTION
**Pull request by berkayoz**
#### Summary

Fixed python packaging, switched missing libraries and updated yaml bits

#### Changes

* Switch from core18 to core20
* Replaced btrfs-tools with btrfs-progs and libbtrfs-dev
* Upgraded libssl1.0.0 to libssl1.1
* Changed python packaging to work with core20

#### Testing

Snapped, installed and tested manually on 18.04, 20.04, 22.04 with workloads.

#### Possible Regressions

Initial assessment seems like the changed packages do not cause a problem, there might be things that hasn't come up yet.

#### Checklist

* [x] Read the [contributions](https://github.com/canonical/microk8s/blob/master/CONTRIBUTING.md) page.
* [x] Submitted the [CLA form](https://ubuntu.com/legal/contributors/agreement), if you are a first time contributor.
* [x] The introduced changes are covered by unit and/or integration tests.

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -177,6 +185,7 @@ parts:
       - git
       - grep
       - hostname
+      - ufw
```
Comment (neoaggelos) [inline snap/snapcraft.yaml:188]: Why do we include ufw in the snap?

Comment (berkayoz) [reply snap/snapcraft.yaml:188]: We are setting the python path environment variable globally in the snap to use the one we ship which is Python 3.8. ufw uses Python, and the snap uses the hosts ufw binary on classic which can lead to errors due to python version mismatch. For example the ufw on 22.04 doesn't seem to be compatible with Python3.8.

So the idea is to include the ufw so we can avoid using the host binary. I'm open to suggestions on how we can handle this.

**Comment by ktsakalozos** [comment]
LGTM but we need to wait to branch out the 1.26 release.

**Comment by ktsakalozos** [comment]
@berkayoz when trying to upload the snap package on the store I got this error:
```
The store was unable to accept this snap.
  - package contains external symlinks: lib/systemd/system/zfs-import.service -> /dev/null, usr/local/bin/iptables-xml -> /usr/local/sbin/xtables-legacy-multi
```

**Comment by berkayoz** [comment]
@ktsakalozos I'm thinking removing [this line](https://github.com/canonical/microk8s/blob/ad8b56bd970c75e59d603df146b6d703919935ea/snap/snapcraft.yaml#L142) could get rid of the last symlink issue, I don't think we do anything special for zfs except staging the `zfsutils-linux` package. I'm not sure how to fix it, why do we need this package? What is depending on it?

**Comment by ktsakalozos** [comment]
@berkayoz for the zfs-import service we can have a 
```
    override-prime: |
      snapcraftctl prime
      rm -vf lib/systemd/system/zfs-import.service
```

But for the iptables-xml I am not sure yet. The iptables deb package is pulled from the apt repo due to `ufw`.  however we want to build the iptables binaries in the dedicated part we have.

**Review by ktsakalozos — APPROVED** [review]
LGTM +1 thank you @berkayoz

**Pull request merged**